File: TourApp.Web/client/projects/shared/models/src/tour/tour-query-settings.ts
import type {OrderByTourProperty, TourDifficulty, TourState} from "./common";


export type TourQuerySettings = {
    id?: string,
    heading?: string,
    remainingPlaces?: string,
    difficulties?: TourDifficulty[],
    tourStates?: TourState[],
    countries?: string[],
    pageIndex?: number,

    "priceSettings.lowerBound"?: string,
    "priceSettings.upperBound"?: string,
    "priceSettings.withDiscount"?: boolean,
    "orderSettings.property"?: OrderByTourProperty,
    "orderSettings.descending"?: boolean
}

File: TourApp.Application/Features/Tours/Specifications/Common/TourQuerySettings.cs
using TourApp.Domain.Entities.Tour.Common;

namespace TourApp.Application.Features.Tours.Specifications.Common;

public record TourQuerySettings
(
    int? Id = null,
    string? Heading = null,
    TourPriceFilter? PriceSettings = null,
    HashSet<TourDifficulty>? Difficulties = null,
    HashSet<TourState>? States = null,
    HashSet<string>? Countries = null,
    TourOrderSettings? OrderSettings = null,
    int? RemainingPlaces = null,
    int? PageIndex = null
);